macro.pfsi/src/Human-PfSI.cpp

```diff
@@ -197,6 +197,27 @@ void human::initialize_movement(){
 
   addEvent2Q(e_move_takeTrip(trip_t,dest_id,this));
 
+  /* vaccinate/treat upon travel */
+  if(tileP->get_patch(dest_id)->get_reservoir()){
+
+    /* vaccine */
+    if(tileP->get_params()->get_travel_vaxx()){
+
+      /* prophylaxis? */
+      bool treat(tileP->get_params()->get_travel_treat());
+      addEvent2Q(e_pfsi_pevaxx(trip_t - 1E-10, treat, this));
+
+    /* no vaccine */
+    } else {
+
+      /* prophylaxis? */
+      if(tileP->get_params()->get_travel_treat()){
+        addEvent2Q(e_pfsi_treatment(trip_t - 1E-10, this));
+      }
+
+    }
+  }
+
```

Comment:
I see the duplication, and it may not feel great, but that's part of the construction here. I feel like, when setting up a time-stepping algorithm, there should be a way to set the state... like the difference between Mealy and Moore machines.